src/views/Main/index.jsx

```diff
@@ -203,34 +220,53 @@ class MainContainer extends Component {
         <div>
           <Header
             selectedTabIndex={selectedTabIndex}
-            handleTabChange={this.handleChangeSelectedTab}
+            handleTabChange={this.handleNavigateAndClear}
           />
           {!userSession && <h3>Please sign in</h3>}
-          {showComponent && (
-            <BugzillaComponentDetails
-              {...showComponent}
-              title={showComponent.title}
-              onGoBack={this.handleComponentBackToMenu}
-            />
-          )}
-          {showPerson && (
-            <PersonDetails
-              person={showPerson}
-              bugzillaComponents={Object.values(bugzillaComponents)}
-              onGoBack={this.handleComponentBackToMenu}
-            />
+          {componentDetails && (
+            <Suspense fallback={<div>Loading...</div>}>
+              <BugzillaComponentDetails
```

Comment:
This should probably be handled inside `BugzillaComponents`.

Reply:
@helfi92 Mind if I push this to another PR?
It's going to require a fair amount of code refactoring and I would like to land this code as it fixes some current navigation issues.

Reply:
Works with me.

Reply:
Filed https://github.com/mozilla/bugzilla-dashboard/issues/72

Thanks!